saleor/graphql/warehouse/dataloaders.py

```diff
@@ -340,3 +340,16 @@ def batch_load(self, keys: Iterable[UUID]) -> List[Optional[Warehouse]]:
             keys
         )
         return [warehouses.get(warehouse_uuid) for warehouse_uuid in keys]
+
+
+class WarehouseByStockIdLoader(DataLoader):
+    context_key = "warehouse_by_stock_id"
+
+    def batch_load(self, keys):
+        stocks = (
+            Stock.objects.using(self.database_connection_name)
+            .filter(pk__in=keys)
+            .select_related("warehouse")
```

Comment:
IMO we should create or use `WarehouseByIdLoader` instead of using `select_related`.
We should avoid using `prefetch_related` or `select_related` instead of that we should reuse other dataloaders.